core/src/com/unciv/ui/utils/ZoomableScrollPane.kt

```diff
@@ -45,4 +49,55 @@ open class ZoomableScrollPane: ScrollPane(null) {
             }
         })
     }
-}
\ No newline at end of file
+
+    override fun getFlickScrollListener(): ActorGestureListener {
+        //This is mostly just Java code from the ScrollPane class reimplemented as Kotlin code
+        //Had to change a few things to bypass private access modifiers
+        return object : ActorGestureListener() {
+            override fun pan(event: InputEvent, x: Float, y: Float, deltaX: Float, deltaY: Float) {
+                setScrollbarsVisible(true)
+                scrollX -= deltaX
+                scrollY += deltaY
+
+                //this is the new feature to fake an infinite scroll
+                when {
+                    continousScrollingX && scrollPercentX >= 1 && deltaX < 0 -> {
+                        scrollPercentX = 0f
+                    }
+                    continousScrollingX && scrollPercentX <= 0 && deltaX > 0-> {
+                        scrollPercentX = 1f
+                    }
+                }
+
+                //clamp() call is missing here but it doesn't seem to make any big difference in this case
+
+                if ((isScrollX && deltaX != 0f || isScrollY && deltaY != 0f)) cancelTouchFocus()
+            }
+
+            //This is the default behavior of the fling implementation
+            override fun fling(event: InputEvent, x: Float, y: Float, button: Int) {
```

Comment:
If this doesn't change anything from the original there should be no reason to override this